tests/components/thethingsnetwork/test_config_flow.py

```diff
@@ -0,0 +1,116 @@
+"""Define tests for the The Things Network onfig flows."""
+
+from ttn_client import TTNAuthError
+
+from homeassistant.components.thethingsnetwork.const import (
+    CONF_API_KEY,
+    CONF_APP_ID,
+    CONF_HOSTNAME,
+    DOMAIN,
+    TTN_API_HOSTNAME,
+)
+from homeassistant.config_entries import SOURCE_REAUTH, SOURCE_USER
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from . import add_schema_suggestion
+from .conftest import API_KEY, APP_ID, CONFIG_ENTRY, HOSTNAME
+
+USER_DATA = {CONF_HOSTNAME: HOSTNAME, CONF_APP_ID: APP_ID, CONF_API_KEY: API_KEY}
+USER_DATA_PARTIAL = {CONF_APP_ID: APP_ID, CONF_API_KEY: API_KEY}
+
+
+async def test_user(
+    hass: HomeAssistant, mock_TTNClient_coordinator, mock_TTNClient_config_flow
+) -> None:
+    """Test user config."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+    )
+
+    assert result["type"] == FlowResultType.FORM
```

Comment:
I recently made a PR to change all the FlowResultType checks to 
```suggestion
    assert result["type"] is FlowResultType.FORM
```

Please update this PR as well